OpenRA.Mods.Common/Widgets/ScrollableLineGraphWidget.cs

```diff
@@ -0,0 +1,570 @@
+
+			if (maxHorizontalOffset > 0)
+				ButtonWidget.DrawBackground(ScrollbarButton, thumbRect, false, thumbPressed, thumbHover, false);
+
+			// Draw arrow decorations
+			var leftOffset = !leftPressed || leftDisabled ? 0 : 1; // Using 1 instead of ButtonDepth for simplicity
+			var rightOffset = !rightPressed || rightDisabled ? 0 : 1;
+
+			var leftArrowImage = getLeftArrowImage.Update((leftDisabled, leftPressed, leftHover, false, false));
+			WidgetUtils.DrawSprite(leftArrowImage,
+				new float2(leftButtonRect.Left + leftOffset, leftButtonRect.Top + leftOffset));
+
+			var rightArrowImage = getRightArrowImage.Update((rightDisabled, rightPressed, rightHover, false, false));
+			WidgetUtils.DrawSprite(rightArrowImage,
+				new float2(rightButtonRect.Left + rightOffset, rightButtonRect.Top + rightOffset));
+
+			// Draw x axis
+			axisFont.DrawTextWithShadow(xAxisLabel,
+				new float2(graphOrigin.X, origin.Y) + new float2(width / 2 - xAxisLabelSize.X / 2, -(xAxisLabelSize.Y + Padding + ScrollbarHeight)),
+				Color.White, BackgroundColorDark, BackgroundColorLight, 1);
+
+			// Draw x axis ticks and labels
+			for (var i = visibleStart; i < visibleEnd; i++)
+			{
+				var screenX = i * xStep + horizontalOffset;
+				if (screenX >= 0 && screenX <= width)
+				{
+					cr.DrawLine(graphOrigin + new float2(screenX, 0), graphOrigin + new float2(screenX, -5), 1, Color.White);
+					if (i % XAxisTicksPerLabel == 0)
+					{
+						var xAxisText = GetXAxisValueFormat().FormatCurrent(i / XAxisTicksPerLabel);
+						var xAxisTickTextWidth = labelFont.Measure(xAxisText).X;
+						var xLocation = screenX - xAxisTickTextWidth / 2;
+						labelFont.DrawTextWithShadow(xAxisText,
+							graphOrigin + new float2(xLocation, 2),
+							Color.White, BackgroundColorDark, BackgroundColorLight, 1);
+					}
+				}
+			}
+
+			// Draw y axis
+			axisFont.DrawTextWithShadow(yAxisLabel,
+				new float2(origin.X, graphOrigin.Y) + new float2(5 - axisFont.TopOffset, -(height / 2 - yAxisLabelSize.X / 2)),
+				Color.White, BackgroundColorDark, BackgroundColorLight, 1, (float)Math.PI / 2);
+
+			for (var y = GetDisplayFirstYAxisValue() ? 0 : yStep; y <= height; y += yStep)
+			{
+				var yValue = y / scale;
+				cr.DrawLine(graphOrigin + new float2(0, -y), graphOrigin + new float2(5, -y), 1, Color.White);
+				var text = GetYAxisValueFormat().FormatCurrent(yValue);
+
+				var textWidth = labelFont.Measure(text);
+
+				var yLocation = y + (textWidth.Y + labelFont.TopOffset) / 2;
+
+				labelFont.DrawTextWithShadow(text,
+					graphOrigin + new float2(-(textWidth.X + 3), -yLocation),
+					Color.White, BackgroundColorDark, BackgroundColorLight, 1);
+			}
+
+			// Bottom line
+			cr.DrawLine(graphOrigin, graphOrigin + new float2(width, 0), 1, Color.White);
+
+			// Left line
+			cr.DrawLine(graphOrigin, graphOrigin + new float2(0, -height), 1, Color.White);
+		}
+
+		public override Widget Clone()
```

Comment:
```suggestion
		public override ScrollableLineGraphWidget Clone()
```

you can now do this in .net 8